src/utils/dnd.js

```diff
@@ -0,0 +1,115 @@
+import { MOUSE_EVENT } from '../const/todo.const';
+
+import { setInsertDirection } from '../store/dnd.store';
+import { removeGuide, updateDropzone } from '../ui/dnd/dnd.dom';
+import { calcPosition } from '../ui/dnd/dnd.style';
+import { cloneDragItem, addDragItem, addGuideLine } from '../ui/dnd/dnd.render';
+
+export class DragAndDrop {
+  id = 0;
+  dragClass = 'is-dragging';
+  guidelineClass = 'drop-guideline';
+  dropzoneClass = 'drop-zone';
+  offsetY = 0;
+  offsetX = 0;
+  insertDirection = null;
+
+  constructor({ buttonClass, draggableClass, dropzone }) {
+    this.draggableClass = draggableClass;
+    this.buttonClass = buttonClass;
+    this.$dropzone = dropzone;
+
+    this.initVar();
+  }
+
+  customEvent() {}
+
+  cancelDragEvent() {
+    this.initVar();
+
+    document.querySelector('.' + this.dragClass)?.remove();
+    removeGuide.bind(this)();
+    this.removeDocumentEvent();
+  }
+
+  //  드래그할 때 위치 이동.
+  moveListItem = (e) => {
+    if (!this.$dragItem) return;
+
+    const $target = e.target;
+
+    calcPosition.bind(this)(e);
+
+    if ($target.classList.contains(this.guidelineClass)) return;
+
+    if ($target.closest('.' + this.dropzoneClass + this.id)) {
```

Comment:
```suggestion
    if ($target.closest(`.${this.dropzoneClass}${this.id}`)) {
```

자바스크립트 템플릿 리터럴을 사용할 수 있습니다

https://caniuse.com/template-literals